docs/source/development/overview.md

```diff
@@ -0,0 +1,203 @@
+   will start with the issue number, which makes branches much easier to track.
+
+1. Check that branch out locally and make commits to it, pushing them to GitHub
+   regularly. Do try and make frequent small commits with clear, specific commit
+   messages.
+
+1. Create a pull request (PR) from the issue branch onto the `develop` branch. The PR
+   description should tag the issue being addressed and explain how the incoming code
+   fixes the issue. You can start a PR as 'draft' PR: this can be a useful way to start
+   describing a PR content and checking that testing is passing before opening a PR up
+   for review.
+
+1. Check that the continuous integration testing passes and fix any issues causing test
+   failures.
+
+1. Request reviews from other package developers. A PR cannot be merged into `develop`
+   until at least one approving review has been added to the code. Reviews will often
+   suggest changes to the code and you should discuss those suggestions and implement
+   them.
+
+1. Once a PR has been approved, the PR can be merged into `develop` and the branch can
+   be deleted.
+
+## The package development environment
+
+The short descriptions below provide the key commands needed to set up your development
+environment and provide links to more detailed descriptions of code development for
+`pyrealm`.
+
+### Python environment
+
+You will need to install Python 3.10 or greater to develop `pyrealm`. We recommend using
+`pyenv` or `pyenv-win` to manage your Python installations but this is only really
+needed if you want to be able to test your code on different python versions.
+
+### Package management
+
+We use [`poetry`](https://python-poetry.org/docs/#installation) for dependency
+management and for managing development environments and you will need to install it.
+
+### Installing `pyrealm`
+
+To develop `pyrealm`, you will also need to install [`git`](https://git-scm.com/) and
+then clone the `pyrealm` GitHub repository.
+
+```sh
+git clone https://github.com/ImperialCollegeLondon/pyrealm.git
+```
+
+You can now use `poetry` to install the package dependencies. This is not just the
+package requirements for end users of the package, but also a wider set of tools used in
+package development. The `poetry` tools uses the
+[pyproject.toml](https://github.com/ImperialCollegeLondon/pyrealm/blob/develop/pyproject.toml)
+file to configure the dependencies that will be installed.
+
+```bash
+poetry install
+```
+
+Poetry uses a virtual environment for package development: all packages are installed to
+a stand-alone python environment that is only used for `pyrealm` development. This makes
+sure that the development environment is consistent across python versions and different
+developers. It does mean you need to put `poetry run` before command to run commands in
+that environment or `poetry shell` to start a new shell that specifically uses that
+environment.
+
```

Comment:
Would it be worth mentioning `poetry update` here?

Reply:
💯 I've added a section on updating and the lock file.